src/freertos_drivers/esp_idf/EspIdfWiFi.hxx

```diff
@@ -0,0 +1,914 @@
+            profile_del(index);
+        }
+        return index;
+    }
+
+    /// Delete a saved WiFi access point profile.
+    /// @param index index of profile to delete, 0xFF removes all
+    /// @return 0 if successful, else -1 if profile not found
+    int profile_del(uint8_t index) override
+    {
+        if (index >= HWDefs::MAX_STA_PROFILES)
+        {
+            return -1;
+        }
+        {
+            OSMutexLock locker(&lock_);
+            memset(userCfg_.sta_[index].ssid_, 0, MAX_SSID_SIZE);
+            memset(userCfg_.sta_[index].pass_, 0, MAX_PASS_SIZE);
+        }
+        config_sync();
+        return 0;
+    }
+
+    /// Get a saved WiFi access point profile by index.
+    /// @param index index within saved profilelist to get
+    /// @param ssid 33 byte array that will return the SSID of the index
+    /// @param sec_type will return the security type of the index
+    /// @param priority will return the priority of the index, may be unused
+    /// @return 0 upon success, -1 on error (index does not exist)
+    int profile_get(int index, char ssid[], SecurityType *sec_type,
+        uint8_t *priority) override
+    {
+        OSMutexLock locker(&lock_);
+        if (index >= HWDefs::MAX_STA_PROFILES)
+        {
+            return -1;
+        }
+        strncpy(ssid, userCfg_.sta_[index].ssid_, MAX_SSID_SIZE);
+        ssid[MAX_SSID_SIZE] = '\0';
+        *sec_type = sec_type_encode(userCfg_.sta_[index].sec_);
+        *priority = 0;
+        return 0;
+    }
+
+protected:
+    /// C structure version of the WiFi configuration settings. These are
+    /// essentially the connection profiles (SSID, password, security type).
+    template<size_t N> struct WiFiConfigNVS
+    {
+        uint32_t magic_; ///< magic number to detect initialization
+        WlanRole mode_; ///< role that the device will operate (AP, STA, etc.)
+        uint8_t padding_[3]; ///< padded for alignment
+        WiFiConfigCredentialsNVS ap_; ///< access point configuration
+        WiFiConfigCredentialsNVS sta_[N]; ///< station profiles
+    }; // template<size_t N> struct WiFiConfigNVS
+    static_assert(sizeof(WiFiConfigNVS<HWDefs::MAX_STA_PROFILES>::mode_) ==
+        sizeof(uint8_t));
+    static_assert(
+        sizeof(WiFiConfigNVS<HWDefs::MAX_STA_PROFILES>) ==
+            4 + 4 + 98 + (98 * HWDefs::MAX_STA_PROFILES),
+        "The size of WiFiConfigNVS has changed. This could lead to "
+        "compatiblity issues in deployed devices.");
+
+    /// cached copy of the wifi user config
+    WiFiConfigNVS<HWDefs::MAX_STA_PROFILES> userCfg_; 
+
+private:
+    /// Get the configured WiFi role.
+    /// @return configured WiFi role
+    WlanRole get_role() override
+    {
+        return userCfg_.mode_;
+    }
+
+    /// Retrieve current access point config, including password.
+    /// @param ssid access point SSID
+    /// @param pass access point password
+    void get_ap_config(std::string *ssid, std::string *pass) override
```

Comment:
why does this get function not return the security type?

Reply:
It is a bit confusing that there is both a public and a nonpublic virtual function called get_ap_config but with different arguments.